### PR TITLE
Add global tab shortcuts.

### PR DESCRIPTION
**Pull request by adamjspooner**
@keybase/react-hackers, this is ready for review. It add global shortcuts for the desktop app's tabs. E.g., `cmd+1` on macOS will navigate you to the people tab.

### CODE REVIEW

File: shared/app/nav.desktop.js [hunk 1]
```diff
@@ -5,26 +5,58 @@ import * as RouteTree from '../route-tree/render-route'
...
 class Nav extends React.Component<Props> {
   render() {
     const {routeSelected, routePath, visibleScreen, layerScreens} = this.props
     return (
       <ErrorBoundary>
         <Box style={stylesTabsContainer}>
-          {routeSelected !== loginTab && <TabBar routeSelected={routeSelected} routePath={routePath} />}
+          {routeSelected !== loginTab && (
+            <TabBar
+              hotkeys={Object.keys(hotKeyTabMap).map(key => `${isDarwin ? 'command' : 'control'}+${key}`)}
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: just make this once in this module, not every time this renders or it'll just keep rerendering as it changes

File: shared/app/nav.desktop.js [hunk 2]
```diff
@@ -53,7 +85,9 @@ const mapStateToProps = (state: TypedState) => ({
   _username: state.config.username,
 })
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({})
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _onHotkey: (cmd: string) => dispatch(navigateTo([hotKeyTabMap[cmd.replace(/(command|control)\+/, '')]])),
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: you should sanity check the value is in the map before dispatching

File: shared/app/nav.desktop.js [hunk 3]
```diff
@@ -5,26 +5,58 @@ import * as RouteTree from '../route-tree/render-route'
...
 
+const hotKeyTabMap: {[string]: Tab} = {
+  '1': peopleTab,
+  '2': chatTab,
+  '3': fsTab,
+  '4': teamsTab,
+  '5': devicesTab,
+  '6': gitTab,
+  '7': settingsTab,
+  '8': walletsTab,
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: you need to use the feature flag for wallets tab

**Comment by adamjspooner** [comment]
@keybase/react-hackers, this is ready for another review.

File: shared/app/nav.desktop.js [hunk 4]
```diff
@@ -5,26 +5,63 @@ import * as RouteTree from '../route-tree/render-route'
...
+const hotKeyTabMap: {[string]: Tab} = {
+  '1': peopleTab,
+  '2': chatTab,
+  '3': fsTab,
+  '4': teamsTab,
+  '5': devicesTab,
+  '6': gitTab,
+  '7': settingsTab,
+}
+if (flags.walletsEnabled) {
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: you can leave this but you could also just do this inline like
```
...(flags.walletsEnabled ? {
'8': walletsTab
} : {})
```

File: shared/app/nav.desktop.js [hunk 5]
```diff
@@ -53,7 +90,13 @@ const mapStateToProps = (state: TypedState) => ({
   _username: state.config.username,
 })
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({})
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _onHotkey: (cmd: string) => {
+    if (hotkeys.includes(cmd)) {
```
Comment (chrisnojima) [inline shared/app/nav.desktop.js]: i meant after you do this substitution
```
const tab = hotKeyTabMap[cmd.replace(/(command|control)\+/, '')]
if (tab) {
...
}

```

**Comment by jzila** [comment]
:tada:

**Comment by adamjspooner** [comment]
@keybase/react-hackers, this is ready for another review.

File: shared/app/nav.desktop.js [hunk 6]
```diff
@@ -5,26 +5,61 @@ import * as RouteTree from '../route-tree/render-route'
...
+  '2': chatTab,
+  '3': fsTab,
+  '4': teamsTab,
+  '5': devicesTab,
+  '6': gitTab,
+  '7': settingsTab,
+  ...(flags.walletsEnabled ? {'8': walletsTab} : {}),
+}
+
+const hotkeys = Object.keys(hotkeyTabMap).map(key => `${isDarwin ? 'command' : 'control'}+${key}`)
```
Comment (akalin-keybase) [inline shared/app/nav.desktop.js]: This didn't work for me on Linux -- I think you want 'ctrl' here

File: shared/app/nav.desktop.js [hunk 7]
```diff
@@ -53,7 +88,14 @@ const mapStateToProps = (state: TypedState) => ({
   _username: state.config.username,
 })
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({})
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _onHotkey: (cmd: string) => {
+    const tab = hotkeyTabMap[cmd.replace(/(command|control)\+/, '')]
```
Comment (akalin-keybase) [inline shared/app/nav.desktop.js]: 'ctrl' here too

File: shared/app/nav.desktop.js (hunk 6 above)
Comment (adamjspooner) [reply shared/app/nav.desktop.js]: Thanks!

**Pull request merged**